modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/source/DefaultSourceCallback.java

```diff
@@ -249,14 +252,50 @@ public void handle(Result<T, A> result, SourceCallbackContext context) {
                                                                                      encodingParam,
                                                                                      mimeTypeInitParam);
 
+    T resultValue = result.getOutput();
+
+    Result<?, ?> decoratedResult = result;
+
+    if (resultValue instanceof InputStream) {
+      decoratedResult = decorateInputStream(result, context);
+    } else if (resultValue instanceof Iterator) {
```

Comment:
this case must account PagingProviders rather than iterators? please check with the sdk guys

Reply:
PagingProviders apply to operations.